packages/server/src/api/routes/tests/couchTestUtils.js

```diff
@@ -64,21 +93,170 @@ exports.createInstance = async (request, appId) => {
 exports.createUser = async (
   request,
   instanceId,
-  username = "bill",
-  password = "bills_password"
+  username = "babs",
+  password = "babs_password"
 ) => {
   const res = await request
     .post(`/api/${instanceId}/users`)
     .set(exports.defaultHeaders)
-    .send({ name: "Bill", username, password })
+    .send({
+      name: "Bill",
+      username,
+      password,
+      accessLevelId: POWERUSER_LEVEL_ID,
+    })
   return res.body
 }
 
+const createUser_WithOnePermission = async (
+  request,
+  instanceId,
+  permName,
+  itemId
+) => {
+  let permissions = await generateAdminPermissions(instanceId)
+  permissions = permissions.filter(
+    p => p.name === permName && p.itemId === itemId
+  )
+
+  return await createUserWithPermissions(
+    request,
+    instanceId,
+    permissions,
+    "onePermOnlyUser"
+  )
+}
+
+const createUser_WithAdminPermissions = async (request, instanceId) => {
+  let permissions = await generateAdminPermissions(instanceId)
+
+  return await createUserWithPermissions(
+    request,
+    instanceId,
+    permissions,
+    "adminUser"
+  )
+}
+
+const createUser_WithAllPermissionExceptOne = async (
+  request,
+  instanceId,
+  permName,
+  itemId
+) => {
+  let permissions = await generateAdminPermissions(instanceId)
+  permissions = permissions.filter(
+    p => !(p.name === permName && p.itemId === itemId)
+  )
+
+  return await createUserWithPermissions(
+    request,
+    instanceId,
+    permissions,
+    "allPermsExceptOneUser"
+  )
+}
+
+const createUserWithPermissions = async (
+  request,
+  instanceId,
+  permissions,
+  username
+) => {
+  const accessRes = await request
+    .post(`/api/${instanceId}/accesslevels`)
+    .send({ name: "TestLevel", permissions })
+    .set(exports.defaultHeaders)
+
+  const password = `password_${username}`
+  await request
+    .post(`/api/${instanceId}/users`)
+    .set(exports.defaultHeaders)
+    .send({
+      name: username,
+      username,
+      password,
+      accessLevelId: accessRes.body._id,
+    })
+
+  const db = new CouchDB(instanceId)
+  const designDoc = await db.get("_design/database")
+
+  const loginResult = await request
+    .post(`/api/authenticate`)
+    .set("Referer", `http://localhost:4001/${designDoc.metadata.applicationId}`)
+    .send({ username, password })
+
+  // returning necessary request headers
+  return {
+    Accept: "application/json",
+    Cookie: loginResult.headers["set-cookie"],
+  }
+}
+
+exports.testPermissionsForEndpoint = async ({
+  request,
+  method,
+  url,
+  body,
+  instanceId,
+  permissionName,
+  itemId,
+}) => {
+  const headers = await createUser_WithOnePermission(
```

Comment:
Please keep naming convention consistent. We use camel case across the entire project